cnf-certification-test/performance/suite.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2020-2023 Red Hat, Inc.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package performance
+
+import (
+	"github.com/onsi/ginkgo/v2"
+	"github.com/sirupsen/logrus"
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/common"
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/identifiers"
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/results"
+	"github.com/test-network-function/cnf-certification-test/internal/crclient"
+	"github.com/test-network-function/cnf-certification-test/pkg/provider"
+	"github.com/test-network-function/cnf-certification-test/pkg/scheduling"
+	"github.com/test-network-function/cnf-certification-test/pkg/testhelper"
+	"github.com/test-network-function/cnf-certification-test/pkg/tnf"
+)
+
+// All actual test code belongs below here.  Utilities belong above.
+var _ = ginkgo.Describe(common.PerformanceTestKey, func() {
+	logrus.Debugf("Entering %s suite", common.PerformanceTestKey)
+	var env provider.TestEnvironment
+	ginkgo.BeforeEach(func() {
+		env = provider.GetTestEnvironment()
+	})
+	ginkgo.ReportAfterEach(results.RecordResult)
+
+	testID, tags := identifiers.GetGinkgoTestIDAndLabels(identifiers.TestSharedCPUPoolSchedulingPolicy)
+	ginkgo.It(testID, ginkgo.Label(tags...), func() {
+		var nonGuaranteedPodContainers = env.GetNonGuaranteedPodContainers()
+		testhelper.SkipIfEmptyAll(ginkgo.Skip, nonGuaranteedPodContainers)
+		testSchedOtherPolicyInSharedCPUPool(&env, nonGuaranteedPodContainers)
+	})
+})
+
+func testSchedOtherPolicyInSharedCPUPool(env *provider.TestEnvironment,
+	nonGuaranteedPodContainers []*provider.Container) {
+	nonCompliantContainers := make(map[*provider.Container][]int)
+	var compliantContainers []*provider.Container
+	for _, testContainer := range nonGuaranteedPodContainers {
+		logrus.Infof("Processing %v", testContainer)
+
+		// Get the pid namespace
+		pidNamespace, err := crclient.GetContainerPidNamespace(testContainer, env)
+		if err != nil {
+			logrus.Errorf("unable to get pid namespace due to: %v", err)
```

Comment:
Do we put the error in the middle of some processing in the claim file? We are not reporting each details of the processing there, rather logging with the DEBUG level.

Reply:
We can put whatever we want in the claim file that can help partners to fix a failing test case. I proposed to do a `continue` because, if you can't get the pid namespace, the other errors that you will get If you keep processing this container won't be very useful. In other workds: what's the point of calling `crclient.GetPidsFromPidNamespace` if you already know that pidNamespace is not valid?